core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala

```diff
@@ -209,32 +204,24 @@ private[ui] class StagePage(parent: StagesTab, store: AppStatusStore) extends We
     val dagViz = UIUtils.showDagVizForStage(stageId, stageGraph)
 
     val currentTime = System.currentTimeMillis()
-    val taskTable = try {
-      val _taskTable = new TaskPagedTable(
-        stageData,
-        UIUtils.prependBaseUri(request, parent.basePath) +
-          s"/stages/stage/?id=${stageId}&attempt=${stageAttemptId}",
-        pageSize = taskPageSize,
-        sortColumn = taskSortColumn,
-        desc = taskSortDesc,
-        store = parent.store
-      )
-      _taskTable
-    } catch {
-      case e @ (_ : IllegalArgumentException | _ : IndexOutOfBoundsException) =>
-        null
-    }
 
     val content =
       summary ++
       dagViz ++ <div id="showAdditionalMetrics"></div> ++
       makeTimeline(
         // Only show the tasks in the table
-        Option(taskTable).map({ taskPagedTable =>
+        () => {
+          val taskDataSource = new TaskDataSource(
```

Comment:
Can we also remove `TaskDataSource`? It seems to be used only in `TaskPagedTable`.